assets/js/googlesitekit/datastore/site/first-party-mode.js

```diff
@@ -0,0 +1,184 @@
+/**
+ * `core/site` data store: first-party-mode.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { isPlainObject } from 'lodash';
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import {
+	commonActions,
+	createRegistrySelector,
+	combineStores,
+} from 'googlesitekit-data';
+import { CORE_SITE } from './constants';
+import { createFetchStore } from '../../data/create-fetch-store';
+
+const SET_FIRST_PARTY_MODE_ENABLED = 'SET_FIRST_PARTY_MODE_ENABLED';
+
+const fetchGetFirstPartyModeSettings = createFetchStore( {
+	baseName: 'getFirstPartyModeSettings',
+	controlCallback: () =>
+		API.get( 'core', 'site', 'fpm-settings', undefined, {
+			useCache: false,
+		} ),
+	reducerCallback: ( state, firstPartyModeSettings ) => {
```

Comment:
It's not essential, but I'd suggest using `createReducer` for the reducers in this store to save all the spread operations...